rusqlite_migration/src/lib.rs

```diff
@@ -600,14 +600,32 @@ impl<'m> Migrations<'m> {
         res
     }
 
-    /// Maximum version defined in the migration set
-    fn max_schema_version(&self) -> SchemaVersion {
-        match self.ms.len() {
-            0 => SchemaVersion::NoneSet,
-            v => SchemaVersion::Inside(
-                NonZeroUsize::new(v).expect("schema version should not be equal to 0"),
-            ),
-        }
+    /// Return the maximum version defined in the migration set.
+    ///
+    /// If you are using [`Migrations::from_directory()`] and `include_dir!`, it is non-trivial to
+    /// access the number of migrations loaded, so you may need to use this method.
+    ///
+    /// For most common scenarios, you should be able to just call
+    /// [`Migrations::to_latest`], which already checks the schema version.
+    ///
+    /// See also: [`Migrations::is_latest_schema_version`].
+    pub fn latest_schema_version(&self) -> usize {
+        self.ms.len()
+    }
+
+    /// Return whether the current database version is the latest one for the loaded migrations.
+    ///
+    /// For most common scenarios, you should be able to just call
+    /// [`Migrations::to_latest`], which already checks the schema version.
+    ///
```

Comment:
If we do decide to add this method, might be worth adding a code example with the if.

Maybe also worth having a full example of an actual backup process to make sure that there is nothing awkward with the lifetimes.